### ISSUE TITLE
wgetpaste also fails with dpaste (suspect 30x redirect failure too)

### ISSUE THREAD

**Issue by mjeveritt** [zlin/wgetpaste#17]
```
[Sun-13-Sep  4:43pm] michael@ds-miche2:pts/6:2074 ~/fabfile/gentoo-install $ wgetpaste -s dpaste packages
Apparently nothing was received. Perhaps the connection failed. Enable --verbose or
--debug to get the output from wget that can help diagnose it correctly.
```
Enabling --verbose proves that the transfer/exchange is fine, but wgetpaste is, again, failing to handle either the redirect or detect the resultant 'answer'/paste correctly.

```
Apparently nothing was received. Perhaps the connection failed.
--2020-09-13 16:43:58--  http://dpaste.com/
Resolving dpaste.com... 35.173.69.207
Connecting to dpaste.com|35.173.69.207|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: /28GU7WLAR [following]
--2020-09-13 16:43:58--  http://dpaste.com/28GU7WLAR
Reusing existing connection to dpaste.com:80.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: https://dpaste.com/28GU7WLAR [following]
--2020-09-13 16:43:59--  https://dpaste.com/28GU7WLAR
Connecting to dpaste.com|35.173.69.207|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10866 (11K) [text/html]
Saving to: 'STDOUT'
```

**Comment by bcattaneo** [comment]
Maybe we can switch this one to JSON as well (https://dpaste.com/api/v2/), what do you think?

**Comment by nvinson** [comment]
> Maybe we can switch this one to JSON as well (https://dpaste.com/api/v2/), what do you think?

As someone who looked into issues like this a year ago, I think this idea is worth a shot.

**Comment by mjeveritt** [comment]
> Maybe we can switch this one to JSON as well (https://dpaste.com/api/v2/), what do you think?

I say go for it too - I will probably find that half the other pastebins are toast over the coming days/weeks, so maybe we can reboot this tool and it might get a few more users again(!).

**Comment by bcattaneo** [comment]
> 
> 
> > Maybe we can switch this one to JSON as well (https://dpaste.com/api/v2/), what do you think?
> 
> I say go for it too - I will probably find that half the other pastebins are toast over the coming days/weeks, so maybe we can reboot this tool and it might get a few more users again(!).

I think that's a good idea. I'll also help try the others during the week, and maybe we can also include `jq` per @nvinson suggestion.

**Comment by mjeveritt** [comment]
yello folks .. any progress on this yet?! :slightly_smiling_face:

**Comment by bcattaneo** [comment]
Hi!
Sorry for the delay. @nvinson did some refactor so we'll see how to continue. I'll re-add those two.

**Comment by mjeveritt** [comment]
@nvinson @Cogitri any chance of merging this? (or its successor)

**Comment by nvinson** [comment]
This one was fixed with commit a933682a5b9b71be0c6b9a6a5f9b0c39b0f69431.  Looks like I forgot to add a fixes line in the commit.

**Comment by nvinson** [comment]
@Cogitri could you close this one?
